### ISSUE TITLE
Implement Ultra Fast Change Detection

### ISSUE THREAD

**Issue by mhevery** [angular/angular#423]
https://docs.google.com/document/d/1QKTbyVNPyRW-otJJVauON4TFMHpl0zNBPkJcTcfPJWg/edit

- [ ] Reflective Mode for Dart/CSP
- [ ] Generation mode for JS (using eval)
- [ ] Transformer for Dart

**Comment by mhevery** [comment]
Brainstorming ideas with @vsavkin 

```
class ChangeDetector {
  valueNames:List;
	  // ::context::
	  // user
		// user.name
		// items
		// items.length
		// items.length == 0
		// [null, 10];
		// items.method(user, 10);
	stubs:List<Function>;
	mementos:List;
  values:List = [null, null, null, null, null];
	opcodes:List;
	  // 
	
	detectChanges() {
	  var newValues = new Array(this.values.length);
		var oldValues = this.values;
		
    for(var pc=0; pc < opcodes.length; pc++) {
		  var opcode = opcodes;
			var frmIdx = opcode & SRC_MASK;
			var toIdx
			var value;
		  switch(opcode) {
			  case PROPERTY_ACCESS:
				  var context = newValue[frmIdx];
			    if (oldValue[toIdx] !== (value = newValue[toIdx] = stubs[toIdx](context)) {
			      oldValue[toIdx] = securityCheck(value);
						var memento = this.mementos[toIdx];
						if (memento) {
							this.dispatcher.notify(memento, value);
						}
			    }
				  break;
			}
		}
	}
}
```

**Comment by mhevery** [comment]
Closed by: 850cf0fef483ed6eaf5dc4731a650b95f4e3753c
